src/GitHub.Exports/Primitives/UriString.cs

```diff
@@ -131,7 +134,9 @@ bool ParseScpSyntax(string scpString)
         /// <returns></returns>
         public Uri ToRepositoryUrl()
         {
-            if (url != null && IsFileUri) return url;
+            // we only want to process urls that represent network resources
+            if (!IsValidUri && !IsScpUri) return url;
+            if (IsValidUri && IsFileUri) return url;
```

Comment:
Actually, nevermind, there's something else in the uri parsing that's making this check more complicated than it should be, let me fix that.